library-model/library-model-generator/src/main/java/com/uber/nullaway/libmodel/LibraryModelGenerator.java

```diff
@@ -0,0 +1,276 @@
+
+    @Override
+    public void visit(ClassOrInterfaceDeclaration cid, Void arg) {
+      /*Currently, this logic only supports @NullMarked annotations on top-level classes and
+      does not handle cases where @NullMarked appears on some nested classes but not others*/
+      cid.getAnnotations()
+          .forEach(
+              a -> {
+                if (a.getNameAsString().equalsIgnoreCase(NULL_MARKED)) {
+                  this.isNullMarked = true;
+                }
+              });
+      super.visit(cid, null);
+    }
+
+    @Override
+    public void visit(MethodDeclaration md, Void arg) {
+      Optional<Node> parentClassNode = md.getParentNode();
+      String parentClassName = "";
+      if (parentClassNode.isPresent()) {
+        if (parentClassNode.get() instanceof ClassOrInterfaceDeclaration) {
+          parentClassName = ((ClassOrInterfaceDeclaration) parentClassNode.get()).getNameAsString();
+        } else if (parentClassNode.get() instanceof EnumDeclaration) {
+          parentClassName = ((EnumDeclaration) parentClassNode.get()).getNameAsString();
+        }
+      }
+      if (this.isNullMarked && hasNullableReturn(md)) {
+        methodRecords.put(
+            packageName
+                + "."
+                + parentClassName
+                + ":"
+                + getMethodReturnTypeString(md)
+                + " "
+                + md.getSignature().toString(),
+            MethodAnnotationsRecord.create(ImmutableSet.of("Nullable"), ImmutableMap.of()));
+      }
+      super.visit(md, null);
+    }
+
+    /**
+     * Determines if a MethodDeclaration can return null.
+     *
+     * @param md The MethodDeclaration instance.
+     * @return {@code true} if the method can return null, {@code false} otherwise.
+     */
+    private boolean hasNullableReturn(MethodDeclaration md) {
+      if (md.getType() instanceof ArrayType) {
+        /* For an Array return type the annotation is on the type when the Array instance is
+        Nullable(Object @Nullable []) and on the node when the elements inside are
+        Nullable(@Nullable Object []) */
+        for (AnnotationExpr annotation : md.getType().getAnnotations()) {
+          if (isAnnotationNullable(annotation)) {
+            return true;
+          }
+        }
+      } else {
+        for (AnnotationExpr annotation : md.getAnnotations()) {
+          if (isAnnotationNullable(annotation)) {
+            return true;
+          }
+        }
+      }
+      return false;
+    }
+
+    /**
+     * Takes a MethodDeclaration and returns the String value for the return type that will be
+     * written into the astubx file.
+     *
+     * @param md The MethodDeclaration instance.
+     * @return The return type string value to be written into the astubx file.
+     */
+    private String getMethodReturnTypeString(MethodDeclaration md) {
+      if (md.getType() instanceof ClassOrInterfaceType) {
+        return md.getType().getChildNodes().get(0).toString();
```

Comment:
Note that using simple names is something we inherited from the CF `stub` format + us not explicitly handling imports, I think. So we might want to change these to FQNs at _some_ point (but probably not in this PR)